hypothesis-python/src/hypothesis/strategies/_internal/types.py

```diff
@@ -480,13 +480,9 @@ def resolve_Type(thing):
     if getattr(args[0], "__origin__", None) is typing.Union:
         args = args[0].__args__
     # Duplicate check from from_type here - only paying when needed.
-    for a in args:  # pragma: no cover  # only on Python 3.6
+    for a in args:
         if type(a) == ForwardRef:
-            raise ResolutionFailed(
-                "thing=%s cannot be resolved.  Upgrading to "
-                "python>=3.6 may fix this problem via improvements "
-                "to the typing module." % (thing,)
-            )
+            raise ResolutionFailed(f"Cannot get type for thing={thing}")
```

Comment:
Let's add a suggestion here to convert `ForwardRef` into a real type. We can probably mention `from __future__ import annotations` as well.

Reply:
Hmm. I'm cautious here because forward refs are often used precisely where you *couldn't* write the type directly, and the future-import might be misleading advice (they should usually resolve before Hypothesis sees them, and it's the default in Python 3.10 onwards).

Reply:
Maybe we can then formulate this problem as `Cannot get forward ref type for thing=`? And it would be up to a reader to decide what to do. 🤔 

I am asking because the current message gives no specifics and I would probably be confused.